### PR TITLE
Clarify verify docs

### PR DESCRIPTION
**Pull request by smith**
These weren't very clear, so added some content to flesh this out and
remove examples that weren't very helpful.

Please fix any bad writing and make this be duplicated in sections it
needs to be.

Signed-off-by: Nathan L Smith <smith@chef.io>

### CODE REVIEW

File: chef_master/source/resource_file.rst [hunk 1]
```diff
@@ -259,9 +259,11 @@ This resource has the following properties:
 ``verify``
    **Ruby Types:** String, Block
 
-   A block or a string that returns ``true`` or ``false``. A string, when ``true`` is executed as a system command.
+   Allows verification of a file's contents before it is created. Creates a temp file and then allows execution of commands or Ruby code to make sure the file meets the needed structure.
```
Comment (thommay) [inline chef_master/source/resource_file.rst]: "a temporary file" seems better here.

Comment (thommay) [inline chef_master/source/resource_file.rst]: "fits the needed structure" doesn't seem to quite capture the intention here, but I'm struggling to come up with better words.
"to ensure the new file is fit for purpose"? I dunno that that's any better.
"the new file is valid"?

**Review by thommay — COMMENTED** [review]
This is definitely a huge step forward, thank you!

File: chef_master/source/resource_file.rst (hunk 1 above)
Comment (smith) [reply chef_master/source/resource_file.rst]: @thommay updated this a little in 5e87a4c.

**Comment by davidwrede** [comment]
gif-keyboard-15326130058755744189

**Pull request merged**